### PR TITLE
allow to c/p config from other place

### LINKED ISSUE #110: use_precommit fails half way through when the repo is not a git repo

### PR DESCRIPTION
**Pull request by lorenzwalthert**
Closes #110 and #85. @pat-s what do you think about the API? As pointed out before, I do not want to use the argument `config` because it conflicts with {precommit}. We can later extend this to work with URLs too. And sorry for the convolution of the PR.

### CODE REVIEW

**Comment by pat-s** [comment]
👍 sounds good, thanks! Checking if the incoming path is a URL and then just reading it should not be a big task.

Maybe the option name is a bit long? `precommit.default_config`? The current name sounds a bit "wordy" to me.

**Comment by lorenzwalthert** [comment]
The option has the same name as the argument in `use_precommit()` for consistency. I agree that it is quite long and wordy. I wanted to avoid `config` because of the API clash with {precommit}. The API also uses the prefix `path` for all arguments that are path related. Though for such a small API, this might be an overkill and we could just use `root` instead of `path_root`. Maybe we should even remove these `path_root` arguments altogether and force people to use `withr::with_dir()` if they really want to apply the commands outside the working directory. Then I wanted to highlight that we do c/p. In sum, this is an overkill. So I think you are right we should change the name. Also, when we allow remote config files, it won't be a path anymore anyways. I'll open a separate issue to discuss the whole API before the CRAN release.

I added support for an URL in 3a94364. I added tests but maybe you can test this also manually once we will have reviewed the API.

**Comment by lorenzwalthert** [comment]
Reference: #112.

**Comment by pat-s** [comment]
Works as expected.

Another goodie would be to see where the remote config was downloaded from:

```r
ℹ Downloading remote config from <info here?>
✔ Sucessfully installed pre-commit for [...]
```

If its a local file (not tested) the message is probably differen? But could also contain the local path.

File: NEWS.md [hunk 1]
```diff
@@ -4,6 +4,7 @@
   config file into the repo at initialization. The argument defaults to 
   `options('precommit.path_cp_config_from')` to make it easy for users to use 
   their preferred hooks in every repo they initialize (#111).
+- Create `r-reticulate` env if not existant before installing into it (#114).
```
Comment (pat-s) [inline NEWS.md]:
```suggestion
- Create `r-reticulate` env if not existent before installing into it (#114).
```

Comment (lorenzwalthert) [reply NEWS.md]: Thanks. For some reason the spell checker hook is not activated...

**Comment by lorenzwalthert** [comment]
True, I did so.

**Pull request merged**